src/esp/gfx/PTexMeshShader.cpp

```diff
@@ -25,226 +25,37 @@ using namespace Magnum;
-        }
-      } else if (p.x < 0) {
-        p.x += tsize;
-        p = RotateUVs(p, rot, tsize);
-
-        adjFace = GetAdjFace(adjFace, (3 - rot) & 3, rot);
-        if (IsValid(adjFace)) {
-          p = RotateUVs(p, rot, tsize);
-          return adjFace;
-        }
-      } else {
-        p = RotateUVs(p, rot, tsize);
-        return adjFace;
-      }
-    }
-  } else {
-    // edge 3
-    if (p.x < 0) {
-      int adjFace = GetAdjFace(faceID, 3, rot);
-      if (IsValid(adjFace)) {
-        p.x += tsize;
-        p = RotateUVs(p, rot, tsize);
-        return adjFace;
-      }
-    }
-    // edge 1
-    else if (p.x > tsize - 1) {
-      int adjFace = GetAdjFace(faceID, 1, rot);
-      if (IsValid(adjFace)) {
-        p.x -= tsize;
-        p = RotateUVs(p, rot, tsize);
-        return adjFace;
-      }
-    }
-  }
-
-  return faceID;
-}
-
-// load texel from atlas, handling adjacent faces
-vec4 texelFetchAtlasAdj(sampler2D tex, int faceID, ivec2 p, int level) {
-  int tsize = tileSize >> level;
-
-  // fetch from adjacent face if necessary
-  faceID = indexAdjacentFaces(faceID, p, tsize);
-
-  // clamp to tile edge
-  p = clamp(p, ivec2(0, 0), ivec2(tsize - 1, tsize - 1));
-
-  ivec2 atlasPos = FaceToAtlasPos(faceID, tsize);
-  return texelFetch(tex, atlasPos + p, level);
-}
-
-// fetch with bilinear filtering
-vec4 textureAtlas(sampler2D tex, int faceID, vec2 p) {
-  int level = 0;
-  p -= 0.5;
-  ivec2 i = ivec2(floor(p));
-  vec2 f = p - vec2(i);
-  return mix(
-      mix(texelFetchAtlasAdj(tex, faceID, ivec2(i), level),
-          texelFetchAtlasAdj(tex, faceID, ivec2(i.x + 1, i.y), level), f.x),
-      mix(texelFetchAtlasAdj(tex, faceID, ivec2(i.x, i.y + 1), level),
-          texelFetchAtlasAdj(tex, faceID, ivec2(i.x + 1, i.y + 1), level), f.x),
-      f.y);
-}
-
-layout(location = 0) out vec4 FragColor;
-uniform sampler2D atlasTex;
-
-uniform float exposure;
-
-in vec2 uv;
-
-void main() {
-  vec4 c = textureAtlas(atlasTex, gl_PrimitiveID, uv * tileSize) * exposure;
-  // c = vec4(1.0f, 1.0f, 1.0f, 1.0f);
-  FragColor = vec4(c.xyz, 1.0f);
-}
-)";
+  MAGNUM_ASSERT_GL_VERSION_SUPPORTED(glVersion_);
 
-PTexMeshShader::PTexMeshShader() {
-  MAGNUM_ASSERT_GL_VERSION_SUPPORTED(GL::Version::GL410);
+  GL::Shader vert{glVersion_, GL::Shader::Type::Vertex};
+  GL::Shader geom{glVersion_, GL::Shader::Type::Geometry};
+  GL::Shader frag{glVersion_, GL::Shader::Type::Fragment};
 
-  GL::Shader vert{GL::Version::GL410, GL::Shader::Type::Vertex};
-  GL::Shader geom{GL::Version::GL410, GL::Shader::Type::Geometry};
-  GL::Shader frag{GL::Version::GL410, GL::Shader::Type::Fragment};
+  const std::string shaderDir = STR(SHADER_DIR);
+  const std::string vertexShaderFile =
+      std::string(shaderDir) + "/" + vertexShaderFilename_;
+  const std::string geometryShaderFile =
+      std::string(shaderDir) + "/" + geometryShaderFilename_;
+  const std::string fragmentShaderFile =
+      std::string(shaderDir) + "/" + fragmentShaderFilename_;
```

Comment:
There's [Utility::Directory::join()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility_1_1Directory.html#aa1c57ae6d2c15c7507ad1ae70810d4de), equivalent to python's `os.path.join()`, that does this more robustly :)